docs/upgrading-to-datastreams.asciidoc

```diff
@@ -0,0 +1,98 @@
+[[upgrade-to-data-streams]]
+=== Upgrade to data streams
+
+* <<why-upgrade-to-data-streams>>
+* <<considerations-data-streams>>
+* <<apm-data-streams-upgrade-steps>>
+* <<apm-data-streams-upgrade-steps-ess>>
+
+Classic indices are deprecated starting in version 7.16.0.
+This means that APM Server will no longer perform any index management -- index templates, ingest pipelines, and ILM policies will all be managed by {fleet}.
+
+Users that are not ready to <<upgrade-to-apm-integration>> should
+upgrade to data streams in the meantime.
+
+[discrete]
+[[why-upgrade-to-data-streams]]
+=== Why upgrade?
+
+Data streams offer a number of benefits over classic indices:
+
+**Data streams**:
+
+* Reduced number of fields per index, better space efficiency, and faster queries
+* More granular data control
+* Errors and metrics data streams are shared with other data sources -- which means better long-term integration with the logs and metrics apps
+* Removes template inheritance for ILM policies and makes use of new {es} index and component templates
+* Fixes <<server-resource-exists-not-alias>>
+
+[discrete]
+[[considerations-data-streams]]
+=== Data streams considerations
+
+* This change cannot be reverted
+* Users on {ece} require additional steps prior to migrating, like configuring TLS certificates for the connection between APM Server and {es}
+
+[discrete]
+[[apm-data-streams-upgrade-steps]]
+=== Upgrade a self-managed APM Server
+
+. <<apm-data-streams-upgrade-1>>
+. <<apm-data-streams-upgrade-2>>
+. <<apm-data-streams-upgrade-3>>
+. <<apm-data-streams-upgrade-4>>
+
+[discrete]
+[[apm-data-streams-upgrade-1]]
+==== Upgrade the Elastic Stack
+
+The Elastic Stack ({es} and {kib}) must be upgraded prior to APM Server.
+See the {stack-ref}/upgrading-elastic-stack.html[Elastic Stack Installation and Upgrade Guide] for guidance.
+
+Review the APM <<release-notes,release notes>>, <<apm-breaking,breaking changes>>,
+and Observability {observability-guide}/whats-new.html[What's new] content for important changes between
+your current APM version and this one.
+
+[discrete]
+[[apm-data-streams-upgrade-2]]
+==== Add the APM integration
```

Comment:
No screenshots here to reduce maintenance costs, but happy to add them if you think they're needed.